main_test.go

```diff
@@ -293,26 +167,21 @@ hi.Person{Name="foo", Age=42}
 --- p.Age: 42
 --- p.Name: foo
 --- p.Work(2)...
-working...
-worked for 2 hours
 --- p.Work(24)...
-working...
-caught: can't work for 24 hours!
+caught: <built-in function hi_Person_Work> returned a result with an error set
 --- p.Salary(2): 20
---- p.Salary(24): caught: can't work for 24 hours!
+--- p.Salary(24): caught: <built-in function hi_Person_Salary> returned a result with an error set
 --- Person.__init__
-caught: invalid type for 'Name' attribute | err-type: <type 'exceptions.TypeError'>
-caught: invalid type for 'Age' attribute | err-type: <type 'exceptions.TypeError'>
-caught: Person.__init__ takes at most 2 argument(s) | err-type: <type 'exceptions.TypeError'>
+caught: argument 2 must be str, not int | err-type: <class 'TypeError'>
+caught: an integer is required (got type str) | err-type: <class 'TypeError'>
+*ERROR* no exception raised!
```

Comment:
yep it apparently doesn't check for *extra* args, just silently ignores them.  it is a bit tricky because not all fields are settable in the constructor so we can't actually check this completely without multiple passes to check etc.  not sure it is worth it.